plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/ExperimentalShippingRecommendation.php

```diff
@@ -89,9 +89,8 @@ public static function has_plugins_active() {
 	 * @return bool
 	 */
 	public static function has_jetpack_connected() {
-		if ( class_exists( '\Jetpack' ) && is_callable( '\Jetpack::is_connection_ready' ) ) {
-			return \Jetpack::is_connection_ready();
-		}
-		return false;
+		$jetpack_connection_manager = new Manager( 'woocommerce' );
+
+		return $jetpack_connection_manager->is_connected() && $jetpack_connection_manager->has_connected_owner();
```

Comment:
Use Jetpack Connection's `Manager` class for managing the connection.

I used the `woocommerce` plugin slug because that's what the WC onboarding uses. This can be anything AFAIK; preferably the plugin being installed but since it isn't installed yet, I thought I should use whatever WC Core uses, which either `woocommerce` or `null`.

In this use case, it might not affect anything.